skore-remote-project/src/skore_remote_project/item/skore_estimator_report_item/metadata.py

```diff
@@ -0,0 +1,106 @@
+from contextlib import suppress
+from inspect import getmembers, ismethod
+
+from joblib import hash
+
+
+def metadata(function):
+    function.metadata = ...
+    return function
+
+
+class Metadata:
+
+    def __init__(self, report):
+        self.report = report
+
+    @metadata
+    def estimator_class_name(self):
+        return self.report.estimator_.__class__.__name__
+
+    @metadata
+    def estimator_hyper_params(self):
+        return {
+            key: value
+            for key, value in self.report.estimator_.get_params().items()
+            if isinstance(value, (type(None), bool, float, int, str))
+        }
+
+    @metadata
+    def dataset_fingerprint(self):
+        return hash(
+            (
+                self.report.X_train,
+                self.report.y_train,
+                self.report.X_test,
+                self.report.y_test,
+            )
+        )
+
+    @metadata
+    def ml_task(self):
+        return self.report.ml_task
+
+    @metadata
+    def metrics(self):
+
+        #
+        # Value:
+        # - ignore list[value] (multi-output)
+        # - ignore {label: value} (multi-class)
+        #
+        # Position: int (to display in parallel coordinates plot) | None (to ignore)
+        #
+
+        def scalar(name, data_source, greater_is_better, position, /):
+            with suppress(AttributeError, TypeError):
+                function = getattr(self.report.metrics, name)
+                value = float(function(data_source=data_source))
+
+                return {
+                    "name": name,
+                    "value": value,
+                    "data_source": data_source,
+                    "greater_is_better": greater_is_better,
+                    "position": position,
+                }
+
+        def fit_time():
```

Comment:
I'm thinking that we should show a "predict_time" that could be `predict_time_test`. We can drop `predict_time_train`.